diff --git a/public/tier0/platform.h b/public/tier0/platform.h
index 73047e1..2ec5af5 100644
--- a/public/tier0/platform.h
+++ b/public/tier0/platform.h
@@ -76,6 +76,9 @@
 #include <unistd.h>
 #include <signal.h>
 #include <time.h>
+# if defined(SUNOS) || defined(__sun)
+#  include <sys/filio.h>
+# endif
 #endif
 
 #ifdef OSX
@@ -565,7 +568,9 @@ typedef void * HINSTANCE;
 //-----------------------------------------------------------------------------
 #if defined( GNUC )
 	#define stackalloc( _size )		alloca( ALIGN_VALUE( _size, 16 ) )
-#if defined(_LINUX) || defined(PLATFORM_BSD)
+#if defined(SUNOS) || defined(__sun)
+	#define mallocsize( _p )	( 0 )
+#elif defined(_LINUX) || defined(PLATFORM_BSD)
 	#define mallocsize( _p )	( malloc_usable_size( _p ) )
 #elif defined(OSX)
 	#define mallocsize( _p )	( malloc_size( _p ) )
