diff --git a/engine/socketcreator.cpp b/engine/socketcreator.cpp
index a6445b9..d458e6b 100644
--- a/engine/socketcreator.cpp
+++ b/engine/socketcreator.cpp
@@ -19,6 +19,9 @@
 #include <netinet/tcp.h>
 #include <errno.h>
 #include <sys/ioctl.h>
+#if defined(SUNOS) || defined(__sun)
+#include <sys/filio.h>
+#endif
 #define closesocket close
 #define WSAGetLastError() errno
 #define ioctlsocket ioctl
