diff --git a/filesystem/filesystem_stdio.cpp b/filesystem/filesystem_stdio.cpp
index 5a8391f..b4172e6 100644
--- a/filesystem/filesystem_stdio.cpp
+++ b/filesystem/filesystem_stdio.cpp
@@ -974,7 +974,7 @@ void CStdioFile::FS_fclose()
 		AUTO_LOCK( m_MutexLockedFD );
 
 		struct _stat buf;
-#ifdef ANDROID
+#if defined(ANDROID) || defined(SUNOS)
 		int fd = fileno( m_pFile ); // need to test this
 #else
 		int fd = fileno_unlocked( m_pFile );
