diff --git a/common/randoverride.cpp b/common/randoverride.cpp
index b3e5bd5..9631a47 100644
--- a/common/randoverride.cpp
+++ b/common/randoverride.cpp
@@ -11,6 +11,8 @@
 // memdbgon must be the last include file in a .cpp file!!!
 #include "tier0/memdbgon.h"
 
+#ifndef SUNOS
+
 #ifdef GNUC
 #define __cdecl 
 #endif
@@ -24,4 +26,6 @@ int __cdecl rand()
 	return RandomInt( 0, VALVE_RAND_MAX );
 }
 
+#endif // SUNOS
+
 #endif // !_STATIC_LINKED || _SHARED_LIB
