diff --git a/mathlib/mathlib_base.cpp b/mathlib/mathlib_base.cpp
index 872ffec..4e788a0 100644
--- a/mathlib/mathlib_base.cpp
+++ b/mathlib/mathlib_base.cpp
@@ -50,10 +50,10 @@ const int nanmask = 255<<23;
 //-----------------------------------------------------------------------------
 // Standard C implementations of optimized routines:
 //-----------------------------------------------------------------------------
-float _sqrtf(float _X)
+float _sqrtf(float x)
 {
 	Assert( s_bMathlibInitialized );
-	return sqrtf(_X); 
+	return sqrtf(x); 
 }
 
 float _rsqrtf(float x)
